File: psra/service/src/main/java/by/arhor/psra/mapper/impl/UserMapperImpl.java
package by.arhor.psra.mapper.impl;

import org.modelmapper.ModelMapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import by.arhor.psra.dto.UserDto;
import by.arhor.psra.repository.model.User;

@Component
public class UserMapperImpl extends AbstractMapper<User, UserDto> {

	@Autowired
	public UserMapperImpl(ModelMapper mapper) {
		super(mapper);
	}

	@Override
	public UserDto mapToDto(User entity) {
		return mapper.map(entity, UserDto.class);
	}

	@Override
	public User mapToEntity(UserDto dto) {
		return mapper.map(dto, User.class);
	}

}

File: psra/service/src/main/scala/by/arhor/psra/mapper/impl/UserMapperImpl.scala
package by.arhor.psra.mapper.impl

import by.arhor.psra.dto.UserDto
import by.arhor.psra.mapper.Mapper
import by.arhor.psra.repository.model.User
import org.modelmapper.ModelMapper
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class UserMapperImpl @Autowired() (private val mapper: ModelMapper) extends Mapper[User, UserDto] {

	override def mapToDto(entity: User): UserDto = mapper.map(entity, classOf[UserDto])

	override def mapToEntity(dto: UserDto): User = mapper.map(dto, classOf[User])

}
